### ISSUE TITLE
Indexes on embedded documents not supported

### ISSUE THREAD

**Issue by kastel** [mongomock/mongomock#213]
Indexes using dot notation are not supported. Instead, a field name with a dot is just interpreted as a literal field. Example:

``` python
import mongomock

coll = mongomock.MongoClient().db.coll

coll.create_index([
    ('sub.a', 1),
], unique=True)

coll.insert_one({'sub': {'a': 1}})

# This shouldn't raise a DuplicateKeyError
coll.insert_one({'sub': {'a': 2}})
```

**Comment by wswld** [comment]
It seems that indexes aren't officially supported at this point:

```
    def create_index(self, key_or_list, cache_for=300, **kwargs):
        if 'unique' in kwargs and kwargs['unique']:
            self._uniques.append(helpers._index_list(key_or_list))

    def drop_index(self, index_or_name):
        pass

    def index_information(self):
        return {}
```

My version is 3.7.0 and these methods look like dummies.

**Comment by pcorpet** [comment]
Thanks for reporting. Mongomock should at least raise a `NotImplementedError` when trying to create a unique index on a nested field.

**Comment by pcorpet** [comment]
This was fixed by @touilleMan in 0b4b6e4b, thanks Emmanuel. It's in Mongomock since v3.11.
